packages/protocol/contracts/governance/Validators.sol

```diff
@@ -1131,4 +1132,30 @@ contract Validators is
       }
     }
   }
+
+  /**
+  * @notice Returns the group that `account` was a member of during `epochNumber`.
+  * @param account The account whose group membership should be returned.
+  * @param epochNumber The epoch number we are querying this account's membership at.
+  * @param index The index into the validator's history struct for their history at `epochNumber`.
+  * @return The group that `account` was a member of during `epochNumber`.
+  */
+  function groupMembershipInEpoch(address account, uint256 epochNumber, uint256 index)
+    external
+    view
+    onlyRegisteredContracts(canForceDeaffiliation)
+    returns (address)
+  {
+    require(isValidator(account) && epochNumber <= getEpochNumber());
+    MembershipHistory storage history = validators[account].membershipHistory;
+    if (
+      history.entries[index].epochNumber == epochNumber ||
+      (history.entries[index].epochNumber < epochNumber &&
+        (history.entries[index.add(1)].epochNumber > epochNumber ||
+          history.entries[index.add(1)].epochNumber == 0))
+    ) {
+      return history.entries[index].group;
+    }
+    require(false, "provided index does not match provided epochNumber at index in history.");
```

Comment:
The more idiomatic approach would be to require that the condition you have in the `if` holds, and then to return `history.entries[index].group;`